libs/wingsdk/src/target-tf-aws/api.cors.ts

```diff
@@ -0,0 +1,146 @@
+import * as cloud from "../cloud";
+
+/**
+ * DEFAULT_RESPONSE is a constant that defines the default response when a request occurs.
+ * It is used to handle all requests that do not match any defined routes in the API Gateway.
+ * The response is a mock integration type, which means it returns a mocked response without
+ * forwarding the request to any backend. The response status code is set to 200 for OPTIONS
+ * and 404 for any other method. The Content-Type header is set to 'application/json'.
+ * @internal
+ */
+export const API_CORS_DEFAULT_RESPONSE = (
+  corsOptions?: cloud.ApiCorsOptions
+) => {
+  if (corsOptions) {
+    return {
+      "/{proxy+}": {
```

Comment:
The cors part works quite well, however the 404 part has issues (see https://github.com/winglang/wing/issues/4091). Likely that it'll have to change to a Lambda function rather than just a mock integration. However, would prefer to do this in another iteration.